packages/shared/src/components/filters/AdvancedSettings.tsx

```diff
@@ -24,31 +32,21 @@ function AdvancedSettingsFilter(): ReactElement {
     });
   };
 
-  useEffect(() => {
-    if (!feedSettings?.advancedSettings) {
-      return;
-    }
-
-    const map = {};
-    feedSettings.advancedSettings.forEach(
-      // eslint-disable-next-line no-return-assign
-      ({ id, enabled }) => (map[id] = enabled),
-    );
-    setSettings(map);
-  }, [feedSettings?.advancedSettings]);
-
   return (
     <section className="flex flex-col px-6" aria-busy={isLoading}>
-      {advancedSettings?.map(({ id, title, description }) => (
-        <FilterSwitch
-          key={id}
-          title={title}
-          name={advancedSettingsKey}
-          description={description}
-          onToggle={() => onToggle(id)}
-          inputId={`${title}-${id}`}
-        />
-      ))}
+      {advancedSettings?.map(
+        ({ id, title, description, defaultEnabledState }) => (
+          <FilterSwitch
+            key={id}
+            label={title}
+            checked={settings[id] ?? defaultEnabledState}
+            name={ADVANCED_SETTINGS_KEY}
```

Comment:
You can't use the same input name for all the components. You can just use the same value like `inputId`